app/controllers/assignments_controller.rb

```diff
@@ -172,7 +173,7 @@ def require_taking_ownership
     return true if Current.user.admin_in?(@roster) || taking_ownership?
 
     flash[:errors] = t('.not_an_admin')
-    redirect_back fallback_location: roster_assignments_path(@roster)
+    redirect_to roster_assignments_path(@roster)
```

Comment:
Now this one needs a bit more thought because it's shared between `create` and `update` and ideally (based on how this is currently set up), we do still want to `render` over `redirect`ing, but with their respective form pages.

The easiest solution would be to parameterize the template we want to render here - i.e.

```ruby
def require_taking_ownership(error_template: nil)
  ...
  flash.now[:error] = t('.not_an_admin')
  render error_template, status: :unprocessable_entity
  ...
end
```
And then in the controller actions
```ruby
require_taking_ownership(error_template: :new) or return
```